src/thirdparty/gpacmp4/mp4.c

```diff
@@ -528,189 +527,255 @@ int processmp4 (struct lib_ccx_ctx *ctx, struct ccx_s_mp4Cfg *cfg, char *file)
 			(unsigned char) ((type>>16)%0x100),(unsigned char) ((type>>8)%0x100),(unsigned char) (type%0x100),
 			(unsigned char) (subtype>>24%0x100),
 			(unsigned char) ((subtype>>16)%0x100),(unsigned char) ((subtype>>8)%0x100),(unsigned char) (subtype%0x100));
-		if (type == GF_ISOM_MEDIA_CAPTIONS || type == GF_ISOM_MEDIA_SUBT || type == GF_ISOM_MEDIA_TEXT)
+		if ((type == GF_ISOM_MEDIA_CAPTIONS && subtype == GF_ISOM_SUBTYPE_C608) ||
+			(type == GF_ISOM_MEDIA_CAPTIONS && subtype == GF_ISOM_SUBTYPE_C708))
 			cc_track_count++;
 		if (type == GF_ISOM_MEDIA_VISUAL && subtype == GF_ISOM_SUBTYPE_AVC_H264)
 			avc_track_count++;
 	}
 
 	mprint("MP4: found %u tracks: %u avc and %u cc\n", track_count, avc_track_count, cc_track_count);
 
-	for (i = 0; i < track_count; i++) {
+	for(i = 0; i < track_count; i++)
+	{
 		const u32 type = gf_isom_get_media_type(f, i + 1);
 		const u32 subtype = gf_isom_get_media_subtype(f, i + 1, 1);
-		mprint("Processing track %d, type=%c%c%c%c subtype=%c%c%c%c\n", i + 1,
-			(unsigned char)(type >> 24 % 0x100), (unsigned char)((type >> 16) % 0x100),
-			(unsigned char)((type >> 8) % 0x100), (unsigned char)(type % 0x100),
-			(unsigned char)(subtype >> 24 % 0x100), (unsigned char)((subtype >> 16) % 0x100),
-			(unsigned char)((subtype >> 8) % 0x100), (unsigned char)(subtype % 0x100));
-
-		const u64 track_type = MEDIA_TYPE(type, subtype);
 
-		switch (track_type) {
-		case MEDIA_TYPE(GF_ISOM_MEDIA_VISUAL, GF_ISOM_SUBTYPE_XDVB): //vide:xdvb
+		if ( type == GF_ISOM_MEDIA_VISUAL && subtype == GF_ISOM_SUBTYPE_XDVB)
+		{
```

Comment:
Is there a problem with MEDIA_TYPE? I think the switch() version is cleaner and having if's.

Reply:
I think so, the problem might be with MEDIA_TYPE, I'll take of the switch() version, here.